### PR TITLE
Checking if content exists when posting to database

### LINKED ISSUE #89: Checking if content exists when posting to database
**Description**
The existence of the content of an image should be checked before putting it in the database.

**Reason**
We need to check if content exists to avoid posting images that don't have one.

**Acceptance criteria**
- [ ] content is being checked before putting an image in the database

### PR DESCRIPTION
**Pull request by mvshao**
**Description**
Fixed bug when the user sends POST request with the invalid content body. API is no longer processing that request.

**Changes**
- API is no longer processing requests with invalid content body

Resolve #89

### CODE REVIEW

**Comment by Halamix2** [comment]
/test all

(https://github.com/kyma-project/test-infra/blob/main/docs/prow/prow-jobs.md#interact-with-prow)

File: backend/internal/api/handler_test.go [hunk 1]
```diff
@@ -369,6 +369,66 @@ func TestCreate(t *testing.T) {
...
+		idMock := mocks.IdGenerator{}
+		idMock.On("NewID").Return(fixedID, nil)
+		eventBusMock := mocks.EventBus{}
+		testSubject := NewHandler(&dbManagerMock, &idMock, &eventBusMock)
+
+		//when
+		testSubject.Create(recorder, req)
+
+		//then
+		dbManagerMock.AssertNumberOfCalls(t, "Get", 0)
```
Comment (mjakobczyk) [inline backend/internal/api/handler_test.go:398]: You can always use `AssertNotCalled` for `0` numbers of calls.

File: backend/internal/api/handler_test.go [hunk 2]
```diff
@@ -369,6 +369,66 @@ func TestCreate(t *testing.T) {
...
```
Comment (mjakobczyk) [inline backend/internal/api/handler_test.go:428]: You can always use `AssertNotCalled` for `0` numbers of calls.

File: backend/internal/api/handler_test.go [hunk 3]
```diff
@@ -369,6 +369,66 @@ func TestCreate(t *testing.T) {
...
 	}
+
+	t.Run("should return error with status code 406 when request content is corrupted", func(t *testing.T) {
+
+		//given
+		hook := test.NewGlobal()
+		img := model.Image{
+			ID:      fixedID,
+			Content: "notAnImage",
+			Time:    time.Now().Format(time.RFC3339),
```
Comment (mjakobczyk) [inline backend/internal/api/handler_test.go:380]: Can't you provide some specific Time object for the testing purposes? Depending on `time.now()` in the unit tests is not considered the best practice.

File: backend/internal/api/handler_test.go [hunk 4]
```diff
@@ -369,6 +369,66 @@ func TestCreate(t *testing.T) {
...
+	})
+
+	t.Run("should return error with status code 500 when content in unmarshaled struct is empty", func(t *testing.T) {
+
+		//given
+		hook := test.NewGlobal()
+		img := model.Image{
+			ID:      fixedID,
+			Content: "",
+			Time:    time.Now().Format(time.RFC3339),
```
Comment (mjakobczyk) [inline backend/internal/api/handler_test.go:410]: Can't you provide some specific Time object for the testing purposes? Depending on `time.now()` in the unit tests is not considered the best practice.

**Review by mjakobczyk — DISMISSED** [review]

**Review by Kaktusowy500 — APPROVED** [review]

**Pull request merged**